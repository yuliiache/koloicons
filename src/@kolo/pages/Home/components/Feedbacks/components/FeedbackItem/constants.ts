export const FEEDBACK_TEXT_TEST_ID = 'feedbackText';
