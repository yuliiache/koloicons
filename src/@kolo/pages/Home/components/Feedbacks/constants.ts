export const FEEDBACKS_KEY = 'feedbacks';
