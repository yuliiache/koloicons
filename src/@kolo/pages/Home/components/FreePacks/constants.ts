export const FREE_PACKS_KEY = 'freePacks';
export const FREE_PACKS_TEST_ID = 'packItemComponent';
