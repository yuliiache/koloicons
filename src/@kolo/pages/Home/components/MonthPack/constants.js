export const MONTH_PACK_KEY = 'monthPack';
