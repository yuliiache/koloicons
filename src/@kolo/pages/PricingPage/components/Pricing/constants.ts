export enum Currency {
  USD = 'USD',
  EUR = 'EUR',
  // Add more as needed
}

export enum SubscriptionPlan {
  PERSONAL = 'Personal',
  BUSINESS = 'Business',
  EXTENDED = 'Extended',
}
