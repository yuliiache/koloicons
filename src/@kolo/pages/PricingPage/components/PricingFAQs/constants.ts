export const PRICING_FAQS_KEY = 'pricingFAQs';
