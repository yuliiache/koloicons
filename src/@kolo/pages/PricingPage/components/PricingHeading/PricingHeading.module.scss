@import "@kolo/design";

.heading {
  padding-top: 16px;
  padding-bottom: 4px;
}

.container {
  padding: 0 16px;
  max-width: 1024px;
  width: 100%;
  margin: 0 auto;

  text-align: center;
}

.title {
  margin: 16px 0 24px;
  font-size: 3.2rem;
  line-height: 1.18;
  font-weight: 900;
  color: $mainFontColor;
}

.introText {
  margin: 0 auto;
  max-width: 480px;
  font-size: 1.2rem;
  line-height: 1.4;
  color: $secFontColor;
}
