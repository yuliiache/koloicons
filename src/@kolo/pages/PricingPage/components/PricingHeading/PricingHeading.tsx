import useNavigateOnSearch from '@kolo/services/hooks/useNavigateOnSearch';
import Search from '@kolo/uiKit/Search/Search';
import {AppRoute} from 'constants/AppRoute';
import {FC} from 'react';

import styles from './PricingHeading.module.scss';

const PricingHeading: FC = () => {
  const {onSearch} = useNavigateOnSearch(AppRoute.ICONS);

  return (
    <section className={styles.heading}>
      <div className={styles.container}>
        <Search
          onSearch={onSearch}
          placeholder="Search icon"
        />
        <h1 className={styles.title}>Pricing</h1>
        <p className={styles.introText}>
          The Biggest Icons Pack. In other words, you will never have to purchase icons again!
        </p>
      </div>
    </section>
  );
};

export default PricingHeading;
