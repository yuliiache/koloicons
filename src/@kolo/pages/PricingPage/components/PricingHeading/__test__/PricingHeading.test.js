import {BrowserRouter} from 'react-router-dom';
import {KeyCode} from '@kolo/constants/constants';
import {SEARCH_BUTTON_WRAP_TEST_ID, SEARCH_INPUT_TEST_ID} from '@kolo/uiKit/Search/constants';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import {AppRoute} from 'constants/AppRoute';

import PricingHeading from '../PricingHeading';

describe('', () => {
  afterEach(cleanup);
  const renderComponent = () =>
    render(
      <BrowserRouter>
        <PricingHeading />
      </BrowserRouter>
    );

  it('should render snapshot successfully', () => {
    const {asFragment} = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass a search query in the pathname and navigate to the icons page when user click the Search button or Enter key', () => {
    const mockInputValue = 'moon';
    const mockPathname = AppRoute.ICONS;
    const mockSearchQuery = `?searchQuery=${mockInputValue}`;

    renderComponent();

    const input = screen.getByTestId(SEARCH_INPUT_TEST_ID);
    const searchButtonWrap = screen.getByTestId(SEARCH_BUTTON_WRAP_TEST_ID);
    const searchButton = searchButtonWrap.querySelector('button');

    fireEvent.change(input, {target: {value: mockInputValue}});
    fireEvent.click(searchButton);
    fireEvent.keyDown(input, {key: KeyCode.ENTER, code: KeyCode.ENTER});

    expect(window.location.pathname + window.location.search).toEqual(mockPathname + mockSearchQuery);
  });
});
