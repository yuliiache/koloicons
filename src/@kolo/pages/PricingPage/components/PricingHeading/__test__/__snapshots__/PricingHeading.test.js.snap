// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render snapshot successfully 1`] = `
<DocumentFragment>
  <section
    class="heading"
  >
    <div
      class="container"
    >
      <div
        class="searchWrap rounded"
      >
        <input
          class="input"
          data-testid="searchInputComponent"
          id="search-input"
          placeholder="Search icon"
          type="text"
          value=""
        />
        <div
          class="searchBtn"
          data-testid="searchButtonComponent"
        >
          <button
            class="button button-square button-big"
            data-testid="buttonIconComponent"
          >
            <svg
              class="iconSearch"
              data-testid="icon-search"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M11,18 C7.13400675,18 4,14.8659932 4,11 C4,7.13400675 7.13400675,4 11,4 C14.8659932,4 18,7.13400675 18,11 C18,14.8659932 14.8659932,18 11,18 Z"
                  fill="#E9EDF7"
                />
                <path
                  d="M11,2 C15.9705627,2 20,6.02943725 20,11 C20,13.124779 19.2636898,15.0775836 18.0323074,16.6171757 L21.7071068,20.2928932 C22.0976311,20.6834175 22.0976311,21.3165825 21.7071068,21.7071068 C21.3165825,22.0976311 20.6834175,22.0976311 20.2928932,21.7071068 L20.2928932,21.7071068 L16.6171757,18.0323074 C15.0775836,19.2636898 13.124779,20 11,20 C6.02943725,20 2,15.9705627 2,11 C2,6.02943725 6.02943725,2 11,2 Z M11,4 C7.13400675,4 4,7.13400675 4,11 C4,14.8659932 7.13400675,18 11,18 C12.8904747,18 14.6059153,17.25059 15.8654514,16.0326404 L15.9428932,15.9428932 C15.9713525,15.914434 16.0011003,15.8880487 16.031949,15.8637373 C17.25059,14.6059153 18,12.8904747 18,11 C18,7.13400675 14.8659932,4 11,4 Z"
                  fill="#717886"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
      <h1
        class="title"
      >
        Pricing
      </h1>
      <p
        class="introText"
      >
        The Biggest Icons Pack. In other words, you will never have to purchase icons again!
      </p>
    </div>
  </section>
</DocumentFragment>
`;
